store/iavl/store.go

```diff
@@ -43,7 +43,14 @@ type Store struct {
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load.
 func LoadStore(db dbm.DB, id types.CommitID, lazyLoading bool) (types.CommitKVStore, error) {
-	tree, err := iavl.NewMutableTree(db, defaultIAVLCacheSize)
+	return LoadStoreWithInitialVersion(db, id, lazyLoading, 0)
+}
+
+// LoadStore returns an IAVL Store as a CommitKVStore setting its initialVersion
+// to the one given. Internally, it will load the store's version (id) from the
+// provided DB. An error is returned if the version fails to load.
+func LoadStoreWithInitialVersion(db dbm.DB, id types.CommitID, lazyLoading bool, initialVersion int64) (types.CommitKVStore, error) {
```

Comment:
We should take a `uint64` here instead of converting/casting, which can panic.

Reply:
I've removed this and placed the casting inside `loadVersion`. Unfortunately, we have to perform at least once cast since the `ver` parameter there has type `int64` while the `LoadStoreWithInitialVersion` method now accepts a `uint64`.